test/e2e-go/features/suspension/suspension_test.go

```diff
@@ -0,0 +1,173 @@
+
+	// wait for n15 to have LastProposed != 0
+	account, err := fixture.LibGoalClient.AccountData(address)
+	for account.LastProposed == 0 {
+		a.NoError(err)
+		a.Equal(basics.Online, account.Status)
+		account, err = fixture.LibGoalClient.AccountData(address)
+		time.Sleep(roundTime)
+	}
+	a.NoError(err)
+
+	// turn off Node15
+	n15, err := fixture.GetNodeController("Node15")
+	a.NoError(err)
+	a.NoError(n15.FullStop())
+
+	afterStop, err := fixture.AlgodClient.Status()
+	a.NoError(err)
+
+	// Advance 60 rounds
+	err = fixture.WaitForRound(afterStop.LastRound+60, 60*roundTime)
+	a.NoError(err)
+
+	// n15account is still online (the node is off, but the account is marked online)
+	account, err = fixture.LibGoalClient.AccountData(address)
+	a.NoError(err)
+	a.Equal(basics.Online, account.Status)
+	voteID := account.VoteID
+	a.NotZero(voteID)
+
+	// Advance 10 more, n15 has been "absent" for 70 rounds now
+	err = fixture.WaitForRound(afterStop.LastRound+70, 15*roundTime)
+	a.NoError(err)
+
+	// n15's account is still online, but only because it has gone "unnoticed"
+	account, err = fixture.LibGoalClient.AccountData(address)
+	a.NoError(err)
+	a.Equal(basics.Online, account.Status)
+
+	fixture.SendMoneyAndWait(70, 1000, 1000, richAccount.Address, address, "")
+	// pay n15, so it gets noticed
+	fixture.SendMoneyAndWait(afterStop.LastRound+70, 5, 1000, richAccount.Address, address, "")
+
+	// n15's account is now offline, but has voting key material (suspended)
+	account, err = fixture.LibGoalClient.AccountData(address)
+	a.NoError(err)
+	a.Equal(basics.Offline, account.Status)
+	a.NotZero(account.VoteID)
+	a.False(account.IncentiveEligible)
+
+	// Use the fixture to start the node again. Since we're only a bit past the
+	// suspension round, it will still be voting.  It should get a chance to
+	// propose soon (15/100 of blocks) which will put it back online.
+	lg, err := fixture.StartNode(n15.GetDataDir())
+	a.NoError(err)
+
+	// Wait for newly restarted node to start. Presumably it'll catchup in
+	// seconds, and propose by round 90
+	_, err = lg.Status()
+	a.NoError(err)
+
+	// Proceed to round 90
+	err = fixture.WaitForRound(90, 20*roundTime)
+	a.NoError(err)
+	// n15's account is back online, with same voting material
+	account, err = fixture.LibGoalClient.AccountData(address)
+	a.NoError(err)
+	a.Equal(basics.Online, account.Status)
+	a.Equal(voteID, account.VoteID)
+	// coming back online by proposal does not make you incentive eligible (you
+	// didn't "pay the fine")
+	a.False(account.IncentiveEligible)
+
+	// but n15 wants incentives, so it keyregs again, paying the extra fee.
+	// We're going to re-reg the exact same key material, so that the running
+	// node can keep voting.
+
+	// we make an _offline_ tx here, because we want to populate the key
+	// material ourself, but copying the accounts existing state. That makes it
+	// an _online_ keyreg.
+	reReg, err := n15c.MakeUnsignedGoOfflineTx(address, 0, 0, 5_000_000, [32]byte{})
```

Comment:
Is expectation that we'll document and show people how to do this? In real life, feel like they'd go back offline and come on again without guidance.

Reply:
We'll certainly need to explain how to go online for incentives in the first place (put in some extra fee).  This particular case should almost never happen though. Once the next PR happens, the heartbeat mechanism ought to bring you back online without requiring a `keyreg`.